src/kfactory/kcell.py

```diff
@@ -4581,12 +4631,576 @@ def _check_cell_ports(p1: Port, p2: Port) -> int:
     return check_int
 
 
+class VKCell(BaseModel, arbitrary_types_allowed=True):
+    """Emulate `[klayout.db.Cell][klayout.db.Cell]`."""
+
+    _ports: Ports
+    _shapes: dict[int, VShapes]
+    _locked: bool
+    _settings: KCellSettings
+    info: Info
+    kcl: KCLayout
+    insts: list[VInstance]
+    _name: str | None
+
+    def __init__(
+        self,
+        name: str | None = None,
+        kcl: KCLayout = kcl,
```

Comment:
Why not do the same trick that we do in KCell?

Reply:
You mean about the name? Because we don't have an index here ;) Nor does it really serve a lot of purpose here. We can have VKCells without names, it will only cause problems if we want to insert it unnamed as a KCell. That case is caught further down.